### ISSUE TITLE
Nix Build Fails on MacOS Big Sur

### ISSUE THREAD

**Issue by tacryt-socryp** [urbit/urbit#4214]
I just upgraded to Big Sur, and have found that while Nix installs properly, Curl fails to build while building the Urbit binary. I believe @matildepark has seen the same issue. Has anyone figured out how to build binaries properly on Big Sur?

```
nix-build -A urbit -A herb --no-out-link
these derivations will be built:
  /nix/store/171adhkwrvqzh02hpc0dl0bpl9xc8mm7-h2o-v2.2.4.drv
  /nix/store/p6bmcia0pxs3y9bw5w2agya8q40mygcq-ed25519-76385f2ebbbc9580a9c236952d68d11d73a6135c.drv
  /nix/store/3f6jdg64hbykzmhh88fvj0qz3fa31wga-ge-additions.drv
  /nix/store/6f8jic884l48r0nynrvq3g5dqnslbd04-ent.drv
  /nix/store/ahw0168a6y6nh786cb4m8is0dpxqkxc0-murmur3-71a75d57ca4e7ca0f7fc2fd84abd93595b0624ca.drv
  /nix/store/hgrsj58z2nw12f0i5ya8izv2sardlqp4-libaes_siv.drv
  /nix/store/k6yma8r932cvxvsjbxpb2j0r2gf69ny1-ivory-header.drv
  /nix/store/nrq2zbh3fl3p0w396yj1627kh4pn2876-libscrypt-029693ff1cbe4f69d3a2da87d0f4f034f92cc0c2.drv
  /nix/store/vfbnampdla31qlll8z9w7478z2q6fxfn-curl-7.72.0.drv
  /nix/store/wfxm8qyb6ag4s7fpvxba9a1xvnxfagyx-softfloat3-ec4c7e31b32e07aad80e52f65ff46ac6d6aad986.drv
  /nix/store/x6v599gsw76fyv2hm6hf30b9ri21gzni-libsigsegv-2.12.drv
  /nix/store/rladyxga5m21agc4y9vr8lkfhx2dwnr3-urbit-1.0.drv
building '/nix/store/vfbnampdla31qlll8z9w7478z2q6fxfn-curl-7.72.0.drv'...
unpacking sources
unpacking source archive /nix/store/ij4z93wmm1ysymb8z9bxhanb5cnz7g0m-curl-7.72.0.tar.bz2
source root is curl-7.72.0
setting SOURCE_DATE_EPOCH to timestamp 1597822714 of file curl-7.72.0/docs/libcurl/libcurl-symbols.3
patching sources
configuring
fixing libtool script ./ltmain.sh
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/40k7ihjsjj6725yaw683ki9xy2dxm7v6-curl-7.72.0 --bindir=/nix/store/ygp3a74mvmxzp47gr65dnwplk06m96lr-curl-7.72.0-bin/bin --sbindir=/nix/store/ygp3a74mvmxzp47gr65dnwplk06m96lr-curl-7.72.0-bin/sbin --includedir=/nix/store/dim2p70srsk616zzs3vhqz9j48agr99z-curl-7.72.0-dev/include --oldincludedir=/nix/store/dim2p70srsk616zzs3vhqz9j48agr99z-curl-7.72.0-dev/include --mandir=/nix/store/3dwc22br9pn6yn4fc17f8kfsvqh30awp-curl-7.72.0-man/share/man --infodir=/nix/store/ygp3a74mvmxzp47gr65dnwplk06m96lr-curl-7.72.0-bin/share/info --docdir=/nix/store/40k7ihjsjj6725yaw683ki9xy2dxm7v6-curl-7.72.0/share/doc/curl --libdir=/nix/store/40k7ihjsjj6725yaw683ki9xy2dxm7v6-curl-7.72.0/lib --libexecdir=/nix/store/40k7ihjsjj6725yaw683ki9xy2dxm7v6-curl-7.72.0/libexec --localedir=/nix/store/40k7ihjsjj6725yaw683ki9xy2dxm7v6-curl-7.72.0/share/locale --without-ca-bundle --without-ca-path --with-ca-fallback --disable-manual --with-ssl=/nix/store/bid8wmf6fb5kg702ywnl7zjzh5158dnr-openssl-1.1.1g-dev --without-gnutls --without-libssh2 --disable-ldap --disable-ldaps --without-libidn --without-brotli
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether make supports nested variables... yes
checking whether to enable debug build options... no
checking whether to enable compiler optimizer... (assumed) yes
checking whether to enable strict compiler warnings... no
checking whether to enable compiler warnings as errors... no
checking whether to enable curl debug memory tracking... no
checking whether to enable hiding of library internal symbols... yes
checking whether to enable c-ares for DNS lookups... no
checking whether to disable dependency on -lrt... (assumed no)
checking whether to enable ESNI support... no
checking for path separator... :
checking for sed... /nix/store/a0zcpf8i0cifrhc4hnp39lzjx03mq3h2-gnused-4.8/bin/sed
checking for grep... /nix/store/j2j2zqinkwb3mgvgap06krb6dv47v7q3-gnugrep-3.5/bin/grep
checking for egrep... /nix/store/j2j2zqinkwb3mgvgap06krb6dv47v7q3-gnugrep-3.5/bin/grep -E
checking for a BSD-compatible install... /nix/store/g77kammvqqn63v01ppig43m2yn2s1lqd-coreutils-8.31/bin/install -c
checking for gcc... clang
checking whether the C compiler works... no
configure: error: in `/private/tmp/nix-build-curl-7.72.0.drv-0/curl-7.72.0':
configure: error: C compiler cannot create executables
See `config.log' for more details
builder for '/nix/store/vfbnampdla31qlll8z9w7478z2q6fxfn-curl-7.72.0.drv' failed with exit code 77
cannot build derivation '/nix/store/rladyxga5m21agc4y9vr8lkfhx2dwnr3-urbit-1.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/rladyxga5m21agc4y9vr8lkfhx2dwnr3-urbit-1.0.drv' failed
make: *** [build] Error 100
```

**Comment by matildepark** [comment]
You knew I was in hell and you *joined me*

**Comment by matildepark** [comment]
I think this is related but I have no idea how to apply the patch in the first comment if so: https://github.com/NixOS/nixpkgs/issues/104580

**Comment by matildepark** [comment]
Possibly moving to staging might help? https://github.com/NixOS/nixpkgs/pull/107013

Don't know how myself, will look into this later today ...

**Comment by thefloweringash** [comment]
Which version of nixpkgs is this? The initial support for Big Sur in (https://github.com/NixOS/nixpkgs/pull/98541) should be enough for most use cases, and has been in `nixpkgs-unstable` for a short while. There's an upcoming change still in `staging` that fixes bootstraping (https://github.com/NixOS/nixpkgs/pull/107013), but it shouldn't matter here.

**Comment by matildepark** [comment]
Didn't mean to summon you, @thefloweringash. I'm afraid I don't know much about Nix — I'm assuming 'stable' on all counts, since all we do is install Nix and then use it for the install here (as you can see in the repo's readme). Is there a quick way to update the install for Big Sur? Updating nix itself didn't change anything.

**Comment by matildepark** [comment]
Do we have to change this? https://github.com/urbit/urbit/blob/6095402687860de1cc54e403fdf5fc0440a8e1f0/nix/sources.json#L98-L108

**Comment by joemfb** [comment]
I've been able to move past this, by locally pointing to the current nixpkgs master as follows:

```
"nixpkgs": {
        "branch": "master",
        "description": "Nix Packages collection",
        "homepage": null,
        "owner": "nixos",
        "repo": "nixpkgs",
        "rev": "29ec4bb3ab38ac0229213197471133770702140e",
        "sha256": "0q73yi4x5sjzkzak589jzwgfw0pd0byrs6lhsx3asrcff7c9afzy",
        "type": "tarball",
        "url": "https://github.com/nixos/nixpkgs/archive/29ec4bb3ab38ac0229213197471133770702140e.tar.gz",
        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
    },
```

**Comment by ryjm** [comment]
been searching for a solution to this for hours, did a double take when i saw your name pop up. 

this is still a problem on ylem too, and updating nixpkgs didn't seem to help

**Comment by joemfb** [comment]
Fixed in #5478
